File: days/1/puzzles/2/program.js
const readline = require('readline');

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
  terminal: false
});

let totalFuelRequired = 0;

rl.on('line', line => {
  if (line) {
    const mass = parseInt(line, 10);

    totalFuelRequired += getFuelRequiredForModuleMass(mass);
  }
});

rl.on('close', () => {
  console.log(totalFuelRequired);
});

function getFuelRequiredForModuleMass(mass) {
  const fuelRequired = Math.floor(mass / 3) - 2;

  if (fuelRequired <= 0) {
    return 0;
  } else {
    return fuelRequired + getFuelRequiredForModuleMass(fuelRequired);
  }
}

File: days/1/puzzles/2/program.py
import sys
import math

total_fuel_required = 0

def get_fuel_required_for_module(mass):
  fuel_required = math.floor(mass / 3) - 2

  if fuel_required <= 0:
    return 0
  else:
    return fuel_required + get_fuel_required_for_module(fuel_required)

for line in sys.stdin:
  module_mass = int(line)

  total_fuel_required += get_fuel_required_for_module(module_mass)

print(total_fuel_required)
